mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -738,6 +797,36 @@ impl MakerSwap {
         };
         swap_events.push(MakerSwapEvent::MakerPaymentInstructionsReceived(instructions));
 
+        let taker_amount = MmNumber::from(self.taker_amount.clone());
+        let remote_version = self
+            .r()
+            .data
+            .taker_version
+            .ok_or("No swap protocol version".to_owned())?;
```

Comment:
shouldn't this be `unwrap_or(0)`? or maybe just not let this be an option and default it to `0` right away.

Reply:
The reason why this taker_version is optional is because it is discovered on the negotiation phase.
I think it is not correct to set it to some specific version on the swap start.
Also, if it is not defined at this stage (wait_taker_fee) - I think this is a clear logic error and probably also not very good to default it to 0 or another version (we should determine it explicitly on the negotiation phase).